MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/LandingActivity.java

```diff
@@ -0,0 +1,126 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.customtabs.CustomTabsIntent;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.github.javiersantos.materialstyleddialogs.MaterialStyledDialog;
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.R;
+
+
+public class LandingActivity extends AppCompatActivity {
+
+  private static final String SIGN_IN_AUTH_URL = "https://api.mapbox.com/oauth/authorize?response_type=code&client_id=%s&redirect_uri=%s";
+
+  private static final String CREATE_ACCOUNT_AUTH_URL = "https://www.mapbox.com/signup/?route-to=https%3A%2F%2Fwww.mapbox"
+    + ".com%2Fauthorize%2F%3Fclient_id%3D7bb34a0cf68455d33ec0d994af2330a3f60ee636%26redirect_uri%3Dmapbox-android-dev-"
+    + "preview%3A%2F%2Fauthorize%26response_type%3Dcode";
+  private static final String CLIENT_ID = "7bb34a0cf68455d33ec0d994af2330a3f60ee636";
+  private static final String REDIRECT_URI = "mapbox-android-dev-preview://authorize";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_landing);
+    setUpButtons();
+    setUpSkipDialog();
+  }
+
+  private void setUpButtons() {
+    Button createAccountButton = (Button) findViewById(R.id.create_account_button);
+    createAccountButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        openChromeCustomTab(true);
+      }
+    });
+
+    Button signInButton = (Button) findViewById(R.id.sign_in_button);
+    signInButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        openChromeCustomTab(false);
+      }
+    });
+  }
+
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+
+    if (getIntent() != null && getIntent().getAction().equals(Intent.ACTION_VIEW)) {
+      Uri uri = getIntent().getData();
+      if (uri.getQueryParameter("error") != null) {
```

Comment:
Why not extracting `uri.getQueryParameter("error")` into a variable `String error` and being able to reuse it when logging (no need to query again)?